lib/event_grid.jl

```diff
@@ -0,0 +1,75 @@
+
+###########################################
+## Interpolated event grid (requires Grid)
+###########################################
+
+@comment """
+# Interpolated event grid
+"""
+
+using Requires
+
+@require Grid begin
+
+    using Grid
+    
+    function search_grid (g::InterpGrid, x::Float64)
+        i = searchsortedfirst (g, x)
+        g[i]
+    end
+
+    function grid_point (x::Float64, v::Float64)
+        x - v
+    end
+    
+@doc* """
+Interpolated event grid (requires Grid library).
+
+```julia
+
+EventGrid(events, tf, dt)
+```
+
+Given a vector with event times and a sampling step, creates a
+continuous interpolated function that has a negative value between
+events, and crosses zero at each event time.
+The function `grid_input` can be used to detect events in model equations.
+
+### Arguments
+
+* `event` : Vector of event times
+* `tf` : grid end time
+* `dt` : event grid sampling time
+
+### Example
+
+```julia
+
+input = EventGrid([1.0,2.0,3.0],tstop,0.1)
```

Comment:
Hi Tom,

  I think EventGrid and grid_input can be merged, I just thought that
having EventGrid as a separate object could potentially give the user more
control over the interpolation method.
But for the time being, it is perfectly fine to merge them. I will then
only export grid_input. I will try to compute some reasonable defaults for
tf and dt as well.

On Sun, Feb 8, 2015 at 5:14 PM, Tom Short <notifications@github.com> wrote:

> In lib/event_grid.jl
> <https://github.com/tshort/Sims.jl/pull/52#discussion_r24305239>:
>
> > +Given a vector with event times and a sampling step, creates a
> > +continuous interpolated function that has a negative value between
> > +events, and crosses zero at each event time.
> > +The function `grid_input` can be used to detect events in model equations.
> > +
> > +### Arguments
> > +
> > +* `event` : Vector of event times
> > +* `tf` : grid end time
> > +* `dt` : event grid sampling time
> > +
> > +### Example
> > +
> > +```julia
> > +
> > +input = EventGrid([1.0,2.0,3.0],tstop,0.1)
>
> Thanks, Ivan. Now for another naive question: could you merge EventGrid
> and grid_input? Then, it'd look like:
>
> Event(eventgrid([1.0,2.0,3.0],tstop,0.1), ...
>
> I'm not sure if these are ever needed independently.
>
> Also, on capitalization, I think these should be lower case. They are not
> types or models.
>
> Lastly, can tf and dt have defaults, possibly based on the range and
> smallest differences of the event times?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tshort/Sims.jl/pull/52/files#r24305239>.
>